In [49]:
%matplotlib inline
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

pd.options.display.float_format = '{:20,.2f}'.format #para arreglar la notacion cientifica

In [28]:
#Usar el area del monitor widescreen
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

# Kickstarter Projects

---
###### Sobre el Proyecto:

Ingestar, explorar, preparar, evaluar y visualizar detalles sobre el exito y fracaso de los proyectos propuestos en **Kickstarter** basado en el set de datos que se encuentra en: https://www.kaggle.com/kemical/kickstarter-projects

###### Que es Kickstarter?

> Kickstarter es una plataforma que provee a artistas, músicos, cineastas, diseñadores y otros creadores con los recursos y el apoyo necesarios para llevar sus ideas a la realidad. 

###### Informacion general de los Proyectos suminstrada por Kickstarter.com:
    Indice de exito actual:                            36,92%
    Contribuciones a proyectos financiados con exito: $3,78 B
    Proyectos activos:                                 3.857

### Ingesta y exploracion:

In [39]:
#Carga de archivos en diferentes dataframes
#data = pd.read_csv('data/ks-projects-201612.csv', low_memory=False) #mismo data set pero con info hasta el 201612
data = pd.read_csv('data/ks-projects-201801.csv', low_memory=False)

##### Prevista de primeras lineas:

In [43]:
data.head()

ID                                               name   \
0  1000002330                    The Songs of Adelaide & Abullah   
1  1000004038                                     Where is Hank?   
2  1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
3  1000011046  Community Film Project: The Art of Neighborhoo...   
4  1000014025                               Monarch Espresso Bar   

        category  main_category  currency             deadline   goal   \
0          Poetry     Publishing       GBP  2015-10-09 11:36:00   1000   
1  Narrative Film   Film & Video       USD  2013-02-26 00:20:50  45000   
2           Music          Music       USD  2012-04-16 04:24:11   5000   
3    Film & Video   Film & Video       USD  2015-08-29 01:00:00  19500   
4     Restaurants           Food       USD  2016-04-01 13:38:27  50000   

             launched  pledged       state  backers  country  usd pledged   \
0  2015-08-11 12:12:28        0      failed        0       GB            0   
1  2013-01-12 00:20:50      220      failed        3       US          220   
2  2012-03-17 03:24:11        1      failed        1       US            1   
3  2015-07-04 08:35:03     1283    canceled       14       US         1283   
4  2016-02-26 13:38:27    52375  successful      224       US        52375   

  Unnamed: 13 Unnamed: 14 Unnamed: 15  Unnamed: 16  
0         NaN         NaN         NaN          NaN  
1         NaN         NaN         NaN          NaN  
2         NaN         NaN         NaN          NaN  
3         NaN         NaN         NaN          NaN  
4         NaN         NaN         NaN          NaN

Nota: *el archivo contiene 4 columnas sin header*

##### Prevista de primeras lineas dataset 2:

In [41]:
data2.head()

ID                                               name  \
0  1000002330                    The Songs of Adelaide & Abullah   
1  1000003930      Greeting From Earth: ZGAC Arts Capsule For ET   
2  1000004038                                     Where is Hank?   
3  1000007540  ToshiCapital Rekordz Needs Help to Complete Album   
4  1000011046  Community Film Project: The Art of Neighborhoo...   

         category main_category currency    deadline     goal  \
0          Poetry    Publishing      GBP  2015-10-09   1000.0   
1  Narrative Film  Film & Video      USD  2017-11-01  30000.0   
2  Narrative Film  Film & Video      USD  2013-02-26  45000.0   
3           Music         Music      USD  2012-04-16   5000.0   
4    Film & Video  Film & Video      USD  2015-08-29  19500.0   

              launched  pledged     state  backers country  usd pledged  \
0  2015-08-11 12:12:28      0.0    failed        0      GB          0.0   
1  2017-09-02 04:43:57   2421.0    failed       15      US        100.0   
2  2013-01-12 00:20:50    220.0    failed        3      US        220.0   
3  2012-03-17 03:24:11      1.0    failed        1      US          1.0   
4  2015-07-04 08:35:03   1283.0  canceled       14      US       1283.0   

   usd_pledged_real  usd_goal_real  
0               0.0        1533.95  
1            2421.0       30000.00  
2             220.0       45000.00  
3               1.0        5000.00  
4            1283.0       19500.00

Nota: *el archivo contiene 2 columnas menos que el anterior y todas las columnas tienen header*

In [42]:
#Detalles del dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 323750 entries, 0 to 323749
Data columns (total 17 columns):
ID                323750 non-null int64
name              323746 non-null object
category          323745 non-null object
main_category     323750 non-null object
currency          323750 non-null object
deadline          323750 non-null object
goal              323750 non-null object
launched          323750 non-null object
pledged           323750 non-null object
state             323750 non-null object
backers           323750 non-null object
country           323750 non-null object
usd pledged       319960 non-null object
Unnamed: 13       625 non-null object
Unnamed: 14       12 non-null object
Unnamed: 15       4 non-null object
Unnamed: 16       1 non-null float64
dtypes: float64(1), int64(1), object(15)
memory usage: 42.0+ MB


Notas:
1. *Las cuatro columnas no identificadas contienen muy pocos valores*
2. *La cantidad de dinero aportado (usd pledged) no se encuentra 100% poblada (3.790 valores nulos)*
3. *Varios objetos que aparentan ser de origen numerico estan identificados como objetos (goal, pledged, backers, usd pledged)*

In [45]:
#Detalles del dataset #2
data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 378661 entries, 0 to 378660
Data columns (total 15 columns):
ID                  378661 non-null int64
name                378657 non-null object
category            378661 non-null object
main_category       378661 non-null object
currency            378661 non-null object
deadline            378661 non-null object
goal                378661 non-null float64
launched            378661 non-null object
pledged             378661 non-null float64
state               378661 non-null object
backers             378661 non-null int64
country             378661 non-null object
usd pledged         374864 non-null float64
usd_pledged_real    378661 non-null float64
usd_goal_real       378661 non-null float64
dtypes: float64(5), int64(2), object(8)
memory usage: 43.3+ MB


Notas:
1. *La cantidad de dinero aportado (usd pledged) no se encuentra 100% poblada (3.797 valores nulos)*
2. *El set de datos contiene las columnas usd_pledged_real y usd_goal_real que fueron obtenidas por una herramienta externa que realizo una conversion de moneda basado en la fecha final del proyecto y que se encuentran 100% pobladas*

In [50]:
#Detalle de los valores numericos
data.describe()

ID           Unnamed: 16
count           323,750.00                 1.00
mean      1,074,942,564.82                25.00
std         619,335,210.13                  nan
min               5,971.00                25.00
25%         537,580,176.00                25.00
50%       1,075,818,379.00                25.00
75%       1,611,070,301.75                25.00
max       2,147,476,221.00                25.00

Notas:
1. *En el set de datos solo 2 campos fueron identificados como de orignen numerico*

In [51]:
#Detalle de los valores numericos
data2.describe()

ID                 goal              pledged  \
count           378,661.00           378,661.00           378,661.00   
mean      1,074,731,191.99            49,080.79             9,682.98   
std         619,086,204.32         1,183,391.26            95,636.01   
min               5,971.00                 0.01                 0.00   
25%         538,263,516.00             2,000.00                30.00   
50%       1,075,275,634.00             5,200.00               620.00   
75%       1,610,148,624.00            16,000.00             4,076.00   
max       2,147,476,221.00       100,000,000.00        20,338,986.27   

                   backers          usd pledged     usd_pledged_real  \
count           378,661.00           374,864.00           378,661.00   
mean                105.62             7,036.73             9,058.92   
std                 907.19            78,639.75            90,973.34   
min                   0.00                 0.00                 0.00   
25%                   2.00                16.98                31.00   
50%                  12.00               394.72               624.33   
75%                  56.00             3,034.09             4,050.00   
max             219,382.00        20,338,986.27        20,338,986.27   

             usd_goal_real  
count           378,661.00  
mean             45,454.40  
std           1,152,950.06  
min                   0.01  
25%               2,000.00  
50%               5,500.00  
75%              15,500.00  
max         166,361,390.71

Notas:
1. *La cantidad de USD promedio para lograr alcanzar el objetivo de un proyecto es de \$45,454*
2. *La candidad promedio de dinero aportardo en un proyecto es de \$9,058*
3. *La cantidad promedio de persona que brindan soporte a un proyecto es de 105*
4. *La maxima cantidad de dinero solicitado para un proyecto es de mas de \$166 millones*
5. *La maxima cantidad de dinero recaudado para un proyecto es de mas de \$20 millones*